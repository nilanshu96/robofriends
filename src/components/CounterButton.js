import { Component } from 'react';

class CounterButton extends Component {

    constructor(props) {
        super(props);
        this.state = {
            count: 0
        }
    }

    shouldComponentUpdate(nextProps, nextState) {
        if (this.state.count !== nextState.count) {
            return true;
        } else {
            return false;
        }
    }

    onCountUpdate = () => {
        this.setState(state => {
            return { count: state.count + 1 }
        })
    }

    render() {
        return (
            <button color={this.props.color} onClick={this.onCountUpdate}>
                Count: {this.state.count}
            </button>
        )
    }
}

export default CounterButton;