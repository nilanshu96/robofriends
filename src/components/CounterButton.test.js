import { render, fireEvent } from '@testing-library/react';
import CounterButton from './CounterButton';

const mockColor = 'red';

test("counter is correct", () => {
    const defaultText = "Count: ";
    const { container, getByRole } = render(<CounterButton color={mockColor} />);
    expect(container.textContent).toBe(defaultText + '0');
    fireEvent.click(getByRole('button'));
    expect(container.textContent).toBe(defaultText + '1');
})

test("CounterButton snapshot test", () => {
    const { container } = render(<CounterButton color={mockColor} />);
    expect(container.innerHTML).toMatchSnapshot();
})