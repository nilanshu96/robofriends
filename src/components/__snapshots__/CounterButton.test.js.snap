// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton snapshot test 1`] = `"<button color=\\"red\\">Count: 0</button>"`;
